pkg/middleware/auth.go

```diff
@@ -98,6 +98,10 @@ func CanAdminPlugins(cfg *setting.Cfg, accessControl ac.AccessControl) func(c *c
 			accessForbidden(c)
 			return
 		}
+		if c.AllowAnonymous && !c.IsSignedIn && shouldForceLogin(c) {
```

Comment:
nit: you can omit `c.AllowAnonymous`, if the user is not logged in and anonymous login is not enabled this code won't be reached and it will be redirected before.